google/cloud/forseti/services/inventory/storage.py

```diff
@@ -129,7 +129,10 @@ def complete(self, status=IndexState.SUCCESS):
             status (str): Final inventory_status.
         """
         self.completed_at_datetime = date_time.get_utc_now_datetime()
-        self.inventory_status = status
+        if self.inventory_index_warnings:
+            self.inventory_status = IndexState.PARTIAL_SUCCESS
```

Comment:
You should set the status when you are calling the complete method instead of updating the status inside the method